2 table.sql

```diff
@@ -0,0 +1,106 @@
+		constraint ck_Passenger_TimeDeparting_cannot_be_more_than_one_year_after_current_date check (TimeDeparting <= dateadd(year, 1, getdate())),
+		--I would add that it cannot be a past date, but the sample data is old
+	TimeArriving datetime not null,
+	PassengerFirstName varchar(30) not null
+		constraint ck_Passenger_PassengerFirstName_cannot_be_blank check (PassengerFirstName > ''),
+	PassengerLastName varchar(30) not null 
+		constraint ck_Passenger_PassengerLastName_cannot_be_blank check (PassengerLastName > ''),
+	PassengerDateOfBirth date not null, 
+	PassengerAddress varchar(150) not null 
+		constraint ck_Passenger_PassengerAddress_cannot_be_blank check (PassengerAddress > ''),
+	PassportNumber char(9) null
+		constraint ck_PassportNumber_must_be_9_numbers_or_null check (PassportNumber like '[1-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9]'),
+	PassportIssueDate date null,
+	PassportExpiryDate date null, 
+	Nationality varchar(30) null,
+	--Is there a point in preventing blanks if I allow nulls?
+	constraint u_Passenger_PassengerFirstName_PassengerLastName_PassengerDateOfBirth_PassengerAdress_and_FlightNumber_must_be_unique 
+		unique (PassengerFirstName, PassengerLastName, PassengerDateOfBirth, PassengerAddress, FlightNumber),
+	constraint ck_Passenger_TimeArriving_must_be_greater_than_TimeDeparting 
+		check (TimeArriving > TimeDeparting),
+	constraint ck_Passport_Passport_Information_must_be_all_null_or_all_entered 
+		check ((PassportNumber is null and PassportIssueDate is null and PassportExpiryDate is null and Nationality is null) or (PassportNumber is not null and PassportIssueDate is not null and PassportExpiryDate is not null and Nationality is not null)),
+		--if I allow blanks for passport info, do I have to add a constraint so that everything is null or blank or everything is entered?
+		--or is it enough to just have a constraint that allows all null or all not null?
+	constraint ck_Passport_DepartureAirport_cannot_be_the_same_as_ArrivalAirport 
+		check (DepartureAirport <> ArrivalAirport),
+	constraint ck_Passenger_Passenger_age_must_be_beween_16_and_90 
+		check (datediff (year, PassengerDateOfBirth, TimeDeparting) between 16 and 90),
+	constraint ck_Passenger_PassportIssueDate_cannot_be_before_of_valid_time_frame
+		check 
+			(
+				(	
+					(datediff(year, PassengerDateOfBirth, PassportIssueDate) >= 16)
+					and
+					(
+						(ArrivalCountry not in (DepartureCountry, Nationality)
+						and TimeDeparting < dateadd(year, 9, dateadd(month, 6,PassportIssueDate)))
+						or
+						ArrivalCountry in (DepartureCountry, Nationality) 
+					)
+				)
+				or 
+				(
+					datediff(year, PassengerDateOfBirth, PassportIssueDate) < 16 
+					and  TimeDeparting < dateadd(year, 5, PassportIssueDate)
+						
+				)
+			),
+	constraint ck_Passenger_PassportIssueDate_must_be_between_PassengerDateOfBirth_and_TimeDeparting check ((PassportIssueDate between PassengerDateOfBirth and TimeDeparting) or PassportIssueDate = ''),
```

Comment:
A date column cannot be ' '. Basically you only want to check if it is valid and if not you just block it and it will be null.

Reply:
took out second part of constraint